tests/integration/helpers/wrap_kakarot.py

```diff
@@ -68,11 +70,43 @@ async def _wrapped(contract, *args, **kwargs):
                     args=args,
                     kwargs=kwargs,
                 )
+
+            codec = Web3().codec
             types = [o["type"] for o in abi["outputs"]]
             data = bytearray(res.result.return_data)
-            decoded = Web3().codec.decode(types, data)
+            decoded = codec.decode(types, data)
             normalized = map_abi_data(BASE_RETURN_NORMALIZERS, types, decoded)
-            return normalized[0] if len(normalized) == 1 else normalized
+            result = normalized[0] if len(normalized) == 1 else normalized
+            log_receipts = [
+                LogReceipt(
+                    address=Web3.toChecksumAddress(f"{evm_contract_address:040x}"),
+                    blockHash=bytes(),
+                    blockNumber=bytes(),
+                    data=bytes(event.data),
+                    logIndex=log_index,
+                    topic=bytes(),
+                    topics=[
+                        bytes.fromhex(
+                            f"{(event.keys[i] + 2**128 * event.keys[i + 1]):064x}"
```

Comment:
could you explain this operation with a comment?

Reply:
see added comment above